### PR TITLE
compare result of evaluated xpath expression to value

### PR DESCRIPTION
**Pull request by pamoller**
See https://github.com/Codeception/Codeception/issues/6546

Evaluate Xpath and compare their values. Youn can also use the xpath count function like checking id is unique:

````php
$this->module->seeResponseJsonXpathEvaluatesTo('count(//id) = 1', true);
````

### CODE REVIEW

File: src/Codeception/Module/REST.php [hunk 1]
```diff
@@ -1100,6 +1100,67 @@ public function seeResponseJsonMatchesXpath(string $xPath): void
...
+     * $I->seeResponseJsonXpathEvaluatesTo('count(//store/book/author) > 0', true);
+     * // count the number of books written by given author is 5
+     * $I->seeResponseJsonMatchesXpath("//author[text() = 'Nigel Rees']", 5);
+     * ```
+     * @part json
+     */
+    public function seeResponseJsonXpathEvaluatesTo(string $xPath, $expected): void
+    {
+        $response = $this->connectionModule->_getResponseContent();
+        $this->assertEquals(
```
Comment (Naktibalda) [inline src/Codeception/Module/REST.php:1142]: Please use `assertSame` instead.

File: src/Codeception/Module/REST.php [hunk 2]
```diff
@@ -1100,6 +1100,67 @@ public function seeResponseJsonMatchesXpath(string $xPath): void
...
+   
+    /**
+     * Opposite to seeResponseJsonXpathEvaluatesTo
+     *
+     * @part json
+     */
+    public function dontSeeResponseJsonXpathEvaluatesTo(string $xPath, $expected): void
+    {
+        $response = $this->connectionModule->_getResponseContent();
+        $this->assertNotEquals(
```
Comment (Naktibalda) [inline src/Codeception/Module/REST.php:1157]: `assertNotSame`

**Review by Naktibalda — CHANGES_REQUESTED** [review]

File: src/Codeception/Module/REST.php (hunk 1 above)
Comment (pamoller) [reply src/Codeception/Module/REST.php:1142]: Sure

File: src/Codeception/Module/REST.php (hunk 2 above)
Comment (pamoller) [reply src/Codeception/Module/REST.php:1157]: sure

**Comment by Naktibalda** [comment]
If count returns `1.0`, then it looks like a good idea to use `assertEquals`

**Comment by Naktibalda** [comment]
In the module too.

**Review by Naktibalda — APPROVED** [review]

**Comment by Naktibalda** [comment]
Released as 3.2.0 https://github.com/Codeception/module-rest/releases/tag/3.2.0

**Pull request merged**